### ISSUE TITLE
substitute rule does not work if the term to substitute is a substring of the preferred term

### ISSUE THREAD

**Issue by obruendl** [jdkato/txtlint#1040]
### Check for existing issues

- [x] Completed

### Environment

Windows 11
Direct download
vale version 3.12.0

### Describe the bug / provide steps to reproduce it

What I want to achieve: The term _gap pad_ shall not be used, instead the term _thermal gap pad_ shall be used.

I have created the following rule:

```
extends: substitution
message: "Preferred '%s' - Found Allowed Term '%s' - Preferred options are 'thermal gap pad, thermal gap pads'"
level: suggestion
ignorecase: false
swap:
  "(?i)\\b(gap\\s*pad|gap\\s*pads)\\b" : "thermal gap pad"
```

This perfectly reports and error for "Is gap pad not reported?".

But it **also does report an unexpected error** for "Is thermal gap pad reported?".

Although I can understand why this happens from a programming perspective, I would not expect this to happen from a user perspective. I'd assume vale tries to match the _biggest token match it can find_ and that the preferred term counts as token as well.

I tried to use a "TokenIgnores=thermal gap pad" in the .vale.ini, but this did not fix the issue (because the term "gap pad" found does not match "thermal gap pad" which Ignore). 

**Can you suggest any workaround to my problem?**

I generate the rules from a database containing preferred and forbidden terms. So I can easily generate whatever workaround rule to NOT report the preferred term.

**Comment by obruendl** [comment]
Is there any update on this one?

The issue persists.

**Comment by jdkato** [comment]
You can exclude the preferred-term context right in the pattern.

```yaml
extends: substitution
message: "Preferred '%s' - use 'thermal gap pad'"
level: suggestion
ignorecase: false
swap:
  "(?i)\\b(?<!thermal\\s)(gap\\s*pads?)\\b": "thermal gap pad"
```

The `(?<!thermal\s)` says "match `gap pad(s)` **unless** it's immediately preceded by `thermal `". So:

- `gap pad` / `Gap Pad` → flagged 
- `thermal gap pad` → not flagged
